modules/apps/portal/portal-verify-extender/src/main/java/com/liferay/portal/verify/extender/internal/osgi/commands/VerifyProcessTrackerOSGiCommands.java

```diff
@@ -176,11 +176,15 @@ protected void activate(
 				public VerifyProcess addingService(
 					ServiceReference<VerifyProcess> serviceReference) {
 
+					boolean initialDeploymentProperty = GetterUtil.getBoolean(
+						serviceReference.getProperty("initial.deployment"));
+
 					VerifyProcess verifyProcess = _bundleContext.getService(
 						serviceReference);
 
-					if (upgrading ||
-						_isInitialDeployment(serviceReference, verifyProcess)) {
+					if ((initialDeploymentProperty &&
```

Comment:
@achaparro this logic is broken.
Before this pull, when upgrading is true, all verify processes must run. Basically we were implying verify follows upgrade, it has always been like this.

With this pull, verify still runs in most cases when "upgrading = true".

Except for this case: initialDeploymentProperty is true, _isInitialDeployment(verifyProcess) is false, and upgrading is true.
This can not be right, even you want to change the behavior of verify must follow upgrade, it should not be changed only for this one case. 

A verify process with "initial.deployment" only means it needs to run on initial deployment. It is still a verify process which should follow the expectation for all verify process, which is running after upgrade.

Verify process must be safe to re-run no matter what, after all user can manually execute it from gogo shell. There is no reason to exclude "initial.deployment" verify process from post upgrading runs.

Reply:
@shuyangzhou I don't agree with that. Initial deployment upgrade processes are the equivalent to initial deployment verify processes, **they only run the first time the module is deployed**. That changed after your changes, that's why I am coming back to the original behavior. Also, that's what we agreed on when we defined it with Product Managerment and GS, need an process that only runs the first time the module is deployed.

This does not mean an initial deployment verify process can not be run when be executed from gogo shell, it will go directly to here with "force" equals true
https://github.com/liferay-core-infra/liferay-portal/blob/738b679ad7286657ed7690b340ec542b8d56dddd/modules/apps/portal/portal-verify-extender/src/main/java/com/liferay/portal/verify/extender/internal/osgi/commands/VerifyProcessTrackerOSGiCommands.java#L222

Reply:
The change from initial deployment upgrade to initial deployment verify is not an equivalent change. It is meant to be executed on initial deployment and after upgrading.

You can not force it to only run for initial deployment. It has 5 existing usages now in master.
```
modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/verify/CommerceAccountServiceVerifyProcess.java
modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/verify/CommercePriceListServiceVerifyProcess.java
modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/verify/CommerceProductServiceVerifyProcess.java
modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/verify/SAPServiceVerifyProcess.java
modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/verify/SXPServiceVerifyProcess.java
```


They existed before this change, they were running when upgrading, they need to still run when upgrading, otherwise this will be changing behavior in a very weird way.

People will be confused why most verify process runs automatically after upgrade and others are not. The bottomline is, they are verify process, they should follow verify process behaviors. They are just taking on an additional duty, which should not affect their original duty.